src/main/java/com/minecolonies/coremod/inventory/CraftingGUIBuilding.java

```diff
@@ -0,0 +1,339 @@
+     */
+    private final InventoryCraftResult craftResult = new InventoryCraftResult();
+
+    /**
+     * The world object.
+     */
+    private final World worldObj;
+
+    private final EntityPlayer player;
+
+    public CraftingGUIBuilding(final InventoryPlayer playerInventory, final World worldIn)
+    {
+        super();
+        this.worldObj = worldIn;
+        this.player = playerInventory.player;
+
+        this.addSlotToContainer(new SlotCrafting(playerInventory.player, this.craftMatrix, this.craftResult, 0, X_CRAFT_RESULT, Y_CRAFT_RESULT)
+        {
+            @Override
+            public boolean canTakeStack(final EntityPlayer playerIn)
+            {
+                return false;
+            }
+        });
+
+        for (int i = 0; i < 2; ++i)
+        {
+            for (int j = 0; j < 2; ++j)
+            {
+                this.addSlotToContainer(new Slot(this.craftMatrix, j + i * 2, X_OFFSET_CRAFTING + j * INVENTORY_OFFSET_EACH, Y_OFFSET_CRAFTING + i * INVENTORY_OFFSET_EACH)
+                {
+                    @Override
+                    public int getSlotStackLimit()
+                    {
+                        return 1;
+                    }
+
+                    @Override
+                    public ItemStack onTake(final EntityPlayer player, final ItemStack stack)
+                    {
+                        return ItemStack.EMPTY;
+                    }
+
+                    @Override
+                    public ItemStack decrStackSize(final int par1)
+                    {
+                        return ItemStack.EMPTY;
+                    }
+
+                    @Override
+                    public boolean isItemValid(final ItemStack par1ItemStack)
+                    {
+                        return false;
+                    }
+
+                    @Override
+                    public boolean canTakeStack(final EntityPlayer par1EntityPlayer)
+                    {
+                        return false;
+                    }
+                });
+            }
+        }
+
+        int i;
+        for (i = 0; i < PLAYER_INVENTORY_ROWS; i++)
+        {
+            for (int j = 0; j < PLAYER_INVENTORY_COLUMNS; j++)
+            {
+                addSlotToContainer(new Slot(
+                        playerInventory,
+                        j + i * PLAYER_INVENTORY_COLUMNS + PLAYER_INVENTORY_COLUMNS,
+                        PLAYER_INVENTORY_INITIAL_X_OFFSET + j * INVENTORY_OFFSET_EACH,
+                        PLAYER_INVENTORY_INITIAL_Y_OFFSET + i * INVENTORY_OFFSET_EACH
+                ));
+            }
+        }
+
+        for (i = 0; i < PLAYER_INVENTORY_COLUMNS; i++)
+        {
+            addSlotToContainer(new Slot(
+                    playerInventory, i,
+                    PLAYER_INVENTORY_INITIAL_X_OFFSET + i * INVENTORY_OFFSET_EACH,
+                    PLAYER_INVENTORY_HOTBAR_OFFSET
+            ));
+        }
+
+        this.onCraftMatrixChanged(this.craftMatrix);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove this call from a constructor to the overridable "onCraftMatrixChanged" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)